lib/shrine/plugins/download_endpoint.rb

```diff
@@ -185,6 +205,11 @@ def get_uploaded_file(serialized)
     rescue Shrine::Error # storage not found
       not_found!
     rescue JSON::ParserError, ArgumentError => error # invalid serialized component
+      if @verifier_secret
```

Comment:
I am not sure if we could add some special formatting / header to know that we should try to verify the data first.
Or should we always try to do that if the `@verifier_secret` is set?